src/clarity-angular/popover/signpost/signpost-trigger.directive.ts

```diff
@@ -40,10 +43,16 @@ export class SignpostTriggerDirective implements OnDestroy {
      * @function onSignpostTriggerClick
      *
      * @description
-     * click handler for the Signpost trigger button used to hide/show SignpostContent.
+     * Enter key, Space key and click handlers for the Signpost trigger button are used to hide/show SignpostContent.
+     *
+     * The enter HostListener handles the case when the directive is on a non-standard element e.g clr-icon without
+     * a button wrapper and we preventDefault to keep the page from scrolling when the Space key is used.
      */
+    @HostListener("keydown.enter", ["$event"])
+    @HostListener("keydown.space", ["$event"])
     @HostListener("click", ["$event"])
-    onSignpostTriggerClick(event: Event): void {
+    onSignpostTriggerClick(event: KeyboardEvent|MouseEvent): void {
         this.ifOpenService.toggleWithEvent(event);
+        event.preventDefault();  // Keeps the page from scrolling down when the signpost is triggered with space
```

Comment:
i remember @youdz mentioning that we cannot do this. Could you please check with him once?

Reply:
Yeah, I was about to mention that `preventDefault` should be called only when the trigger is activated by clicking the space key. But now I think it shouldn't be called at all. Otherwise, if users use a checkbox or radio button as a signpost trigger, they wouldn't be checked or unchecked.

Reply:
Ok. I'll revisit adding support for Enter/Space keys. Maybe I should limit it to Enter for now?